torch/_inductor/fx_passes/fuse_attention.py

```diff
@@ -453,6 +453,49 @@ def _sfdp_replacement_17(query, key, value, attn_mask, inv_scale, dropout_p):
     )
 
 
+def _sfdp_pattern_18(
+    query, key, value, inv_scale, causal_mask_value, causal_mask, dropout_p
+):
+    # for hf_GPT2 with dropout (introduces clone node) for inference
+    # For inference, it also returns permuted key & value
+    query = query.permute([0, 2, 1, 3])
+    key = key.permute([0, 2, 1, 3])
+    value = value.permute([0, 2, 1, 3])
+    attn_weights = torch.matmul(query, key.permute(0, 1, 3, 2))
+    attn_weights = attn_weights.div(inv_scale)
+    attn_weights = torch.where(causal_mask, attn_weights, causal_mask_value)
+    return (
+        (
+            torch.nn.functional.dropout(attn_weights.softmax(dim=-1), dropout_p).matmul(
+                value
+            )
+        ),
+        key,
+        value,
+    )
+
+
+def _sfdp_replacement_18(
+    query, key, value, inv_scale, causal_mask_value, causal_mask, dropout_p
```

Comment:
For `causal_mask_value`, SDPA only accepts `-inf`. Is it better to use `causal_mask_value` as a constant rather than an input, because this may cause accuracy issue? Similar as https://github.com/pytorch/pytorch/blob/main/torch/_inductor/fx_passes/fuse_attention.py#L425.

Reply:
Thanks for the info! I'm not using the value of `causal_mask_value` as a parameter in `torch.nn.functional.scaled_dot_product_attention`, but it was necessary to match the pattern.

In GPT2, `causal_mask_value` is `torch.finfo(dtype).min`, but the output of the SDPA op should be similar to it being `-float("inf")`, so I'm using the latter in the replacement pattern, similarly as the reference you provided.

I'll apply the other changes.

Reply:
I mean that we could remove `causal_mask_value` in the parameter lists and do `causal_mask_value = torch.finfo(dtype).min` in the pattern function.

Reply:
Thanks, but IIRC, you had a comment in a PR about the pattern not matching when `torch.finfo(dtype).min` was used. I'll try it, and will push changes, if it'd work.

Reply:
Do you indicate this [issue](https://github.com/pytorch/pytorch/issues/117735), in which the problem is `torch.tensor()`? I suppose `torch.finfo(scores.dtype).min` should work and you could have a try. Thanks~

Reply:
It did not match when `torch.finfo(dtype).min` was used. Should I add a comment there?

Reply:
Could you please elaborate on why `torch.finfo(dtype).min` cannot be matched? Is it the same issue as https://github.com/pytorch/pytorch/issues/117735?

Reply:
Sorry, I had forgotten to add a `torch.full` node while I was testing. The pattern matched.